app/controllers/admin_controller.rb

```diff
@@ -0,0 +1,13 @@
+class AdminController < ApplicationController
+  before_action :require_admin
+
+  def show; end
+
+private
+
+  def require_admin
+    unless current_user.admin?
+      redirect_to sign_in_path
+    end
+  end
```

Comment:
If the user is signed in and not admin, should we show them 'forbidden' or something like that instead of asking them to sign in again?

Reply:
Ok the flow is the same, but I've used a flash and now adds a banner to the sign in page with some copy. Does that improve things a little?